### PR TITLE
Simplify c2rust (with latest C2Rust version)

### PR DESCRIPTION
**Pull request by chrysn**
This is equivalent to #196 (also fixing https://github.com/RIOT-OS/RIOT/pull/18056) except that it is

* rebased already, and
* uses the latest upstream version after upstream merged all my relevant pull requests (they're still a bit slow on doing a Cargo release, but installing from a rev in their git history is actually easy).

(Opened as a separate PR because if this fails for some reason, it's easy to fall back to #196).

### CODE REVIEW

**Comment by chrysn** [comment]
Build failure is being investigated, looks like they broke something in the Cargo.toml workspace setup -- it's a build system issue.

**Comment by chrysn** [comment]
Downgraded C2Rust in a hotfix again by a bit: Starting [a commit later](https://github.com/immunant/c2rust/commit/0f5a0943b29b14c9162f13b9a9a63a08ab4a81f8) it is using a more recent Rust version that we currently ship (the latest stable) -- not a problem to upgrade, but not a thing I'd want to tangle in here either.

**Comment by chrysn** [comment]
nagh ... and now *something* is leaving root-owned files in /data/riotbuild; debugging...

File: riotbuild/Dockerfile [hunk 1]
```diff
@@ -309,6 +312,12 @@ RUN \
         true; \
     done"
 
+RUN \
+    echo 'Installing C2Rust' >&2 && \
+    CARGO_HOME=/opt/rustup/.cargo cargo install --no-track --locked c2rust --git https://github.com/immunant/c2rust --rev ed56c794230bbf5a0e417941e4778316d31ab114 && \
+    echo 'Cleaning up root-owned crates.io cache' >&2 && \
```
Comment (aabadie) [inline riotbuild/Dockerfile:318]: Maybe set the rustup version in an ARG ?

```suggestion
ARG RUSTUP_VERSION=ed56c794230bbf5a0e417941e4778316d31ab114
RUN \
    echo 'Installing C2Rust' >&2 && \
    CARGO_HOME=/opt/rustup/.cargo cargo install --no-track --locked c2rust --git https://github.com/immunant/c2rust --rev $RUSTUP_VERSION && \
    echo 'Cleaning up root-owned crates.io cache' >&2 && \
```

Comment (chrysn) [reply riotbuild/Dockerfile:318]: I expect the next change on this to be switching from a commit version to a cargo released version, which would then be a different line (no --git, no --rev) again. Then it'd make sense, now I see little value in it.

Comment (aabadie) [reply riotbuild/Dockerfile:318]: Ok I see, let's leave it like this then!

**Review by kaspar030 — APPROVED** [review]
ACK.

**Comment by kaspar030** [comment]
bors merge

**Pull request merged**